handlers.go

```diff
@@ -609,3 +644,16 @@ func getImg() image.Image {
 	}
 	return img
 }
+
+func parseData(r *http.Request) ([]byte, error) {
+	if r.Body == nil {
+		return []byte{}, nil
+	}
+
```

Comment:
add here a `defer r.Body.Close()` after `r.Body!=nil` check

Reply:
We need to do this to make sure we close the reader if an error happens while we `ioutil.ReadAll()`,  right?  Because otherwise, I think `ReadAll()` closes the file.

Reply:
https://golang.org/pkg/io/ioutil/#ReadAll doesn't seem like this closes. You should close it even if it succeeds. Just make sure `r.Body` is not nil, so my first comment is accurate.